example/shared/src/commonMain/kotlin/viewmodel/system/network/NetworkViewModel.kt

```diff
@@ -0,0 +1,64 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.example.shared.viewmodel.system.network
+
+import com.splendo.kaluga.architecture.observable.toObservable
+import com.splendo.kaluga.architecture.viewmodel.BaseViewModel
+import com.splendo.kaluga.base.flow.HotFlowable
+import com.splendo.kaluga.system.network.Network
+import com.splendo.kaluga.system.network.state.NetworkStateRepoBuilder
+import com.splendo.kaluga.system.network.state.network
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.launch
+
+class NetworkViewModel(
+    networkStateRepoBuilder: NetworkStateRepoBuilder
+) : BaseViewModel() {
+
+    private val networkRepo = networkStateRepoBuilder.create()
+
+    private val _networkState: HotFlowable<String?> = HotFlowable(null)
```

Comment:
Lets just use MutableStateFlow since I think HotFlowable is gonna disappear with the changes made by @thoutbeckers

Reply:
Yes it will be deprecated in my PR.

Aside from the existing stateflow there will be sharedflow observables, but this is clearly state I think.

Reply:
@thoutbeckers in order to use`StateFlow` shouldn't we update coroutines version to 1.4?

Reply:
No, I think you're confused with `SharedFlow`.

But we will upgrade that anyway.

Reply:
4c787bd